### PR TITLE
New Message Queue Patch

### PR DESCRIPTION
**Pull request by zealousduck**
This patch has major restructuring to program functionality to reflect Wade's new requirement for functionality.

Supervisor now calls upon Reader to scan the camera for images first (ReaderScanPatch implementation). Once scanned, the Handler is started to process those newly-downloaded images, comparing to a list of old images (stored in a file.txt). Any detected images are enqueued as normal.

Reader, Handler, and Uploader no longer operate in infinite loops. Instead, they run exactly once whenever the corresponding .run() methods. They will terminate and need to be restarted.

Additionally, the message queues have been implemented bi-directionally, allowing Children to communicate with Supervisor, and allowing the Supervisor to send display messages for the TouchScreenGUI.

### CODE REVIEW

**Comment by zealousduck** [comment]
Note: This should hook directly into the GUI that has the gear graphic, or really any other GUI that is compatible with the interpreting string conditions.

Note: ReaderScanPatch will hook into the Reader.run() method that exists here. Should be pretty painless.

File: production/Handler.py [hunk 1]
```diff
@@ -22,30 +22,67 @@ def __init__(self, orderQueue):
         self.orders = orderQueue
         self.uploadOrders = Queue()
         self.queue = Queue()
-        self.imageList = self.getDirectoryList(self.directoryName)
+        self.listFileName = Utility.UPLOADS_FILE_NAME
+        self.exitMessage = ""
+        
+#     def run(self):
```
Comment (zealousduck) [inline production/Handler.py:28]: Old version of run. Maintaining for posterity currently, will eventually remove these comments.

File: production/PhotoUploadUtility.py [hunk 2]
```diff
@@ -7,12 +7,23 @@
 '''
 
 CONFIG_FILE_NAME = 'photoUpload.cfg'
+UPLOADS_FILE_NAME = 'imageList.txt'
 DEFAULT_CONFIG = 'photoUploadDefaults.cfg'
-POLL_TIME = 3   # in seconds
+POLL_TIME = 1   # in seconds
+QMSG_SETTINGS = 'msg_settings'
```
Comment (zealousduck) [inline production/PhotoUploadUtility.py:13]: QMSG constants used for the various messages the components are passing around.

File: production/Reader.py [hunk 3]
```diff
@@ -21,8 +21,9 @@ def __init__(self, orders):
     
     def run(self):
         print "Hi, I'm a Reader!"
-
-        time.sleep(Utility.POLL_TIME)
+        status = Utility.readMessageQueue(self.myOrders)
+        if status == Utility.QMSG_SCAN:
+            pass # READER SCAN PATCH INSERTED LOGIC HERE
```
Comment (zealousduck) [inline production/Reader.py:26]: See ReaderScanPatch integration!

**Comment by emseduau** [comment]
looks good, I know a bunch of what I worked on is messy and I'll devote some time to refactoring it shortly.

**Pull request merged**